src/server/req_runjob.c

```diff
@@ -340,11 +340,26 @@ req_runjob(struct batch_request *preq)
 
 #ifndef NAS /* localmod 133 */
 	if ((psched->scheduler_sock != -1) && was_job_alteredmoved(parent)) {
-		int index = find_attr(sched_attr_def, ATTR_throughput_mode, SCHED_ATR_LAST);
-		/* do not blacklist altered/moved jobs when throughput_mode is enabled */
-		if ((index == -1) ||
-			((psched->sch_attr[index].at_flags & ATR_VFLAG_SET) &&
-			 (psched->sch_attr[index].at_val.at_long == 0))) {
+		int index_tp = find_attr(sched_attr_def, ATTR_throughput_mode, SCHED_ATR_LAST);
```

Comment:
I don't see any purpose for this find_attr. Attribute array is static, can't you directly index it using SCHED_ATR_job_run_wait or SCHED_ATR_throughput_mode?

Reply:
you are right, I'd just extrapolated the existing code, we should be able to access the array directly.